tests/lib/rules/header.js

```diff
@@ -127,6 +127,13 @@ ruleTester.run("header", rule, {
             errors: [
                 {message: "incorrect header"}
             ]
+        },
+        {
+            code: "/* Copyright 20\n Author: abc@example.com */",
+            options: ["block", {pattern: "^\\s*Copyright \\(c\\) \\d{4} ABC Inc. All rights reserved.\\s*$"}, "Copyright information error. Please add below line to your file.\n\tCopyright (c) " + new Date().getFullYear() + " ABC Inc. All rights reserved"],
+            errors: [
+                {message: "Copyright information error. Please add below line to your file.\n\tCopyright (c) " + new Date().getFullYear() + " ABC Inc. All rights reserved"}
```

Comment:
These tests will fail in 2018.

Reply:
@Stuk, maybe I am missing something, why would this fail in 2018? 

The `pattern` does not match the `code`, which would cause it to throw an error. And the supplied error message would match the output since `new Date().getFullYear()` is used in both places.

Reply:
@Stuk do you mean it will fail in 2100 ?